_pytest/assertion/rewrite.py

```diff
@@ -215,7 +215,8 @@ def load_module(self, name):
             mod.__loader__ = self
             py.builtin.exec_(co, mod.__dict__)
         except:
-            del sys.modules[name]
+            if name in sys.modules:
```

Comment:
This looks good, I think we should add a test for it though.

I tried this test on `master`, but it passes:

```python
pytest_plugins = ['pytester']

def test_import_error(testdir):
    testdir.syspathinsert()
    testdir.makepyfile(top_module='''
        try:
            import other_module
        except ImportError:
            pass
    ''')
    testdir.makepyfile('''
        def test_import_error():
            import top_module
            assert 1
    ''')
    r = testdir.runpytest()
    r.stdout.fnmatch_lines(['* 1 passed *'])
```

Perhaps `anydbm` is doing some more magic behind the scenes?